doc/melange.rst

```diff
@@ -0,0 +1,298 @@
+    ``(modes byte native melange)``
+
+- ``(melange.runtime_deps <deps>)``: optionally, define any runtime dependencies
+  using ``melange.runtime_deps``. This field is analog to the ``runtime_deps``
+  field used in ``melange.emit`` stanzas. An example:
+  ``(melange.runtime_deps ./path/to/file.css (glob_file_rec ./fonts/*))``.
+
+Recommended Practices
+=====================
+
+Keep Bundles Small by Reducing the Number of ``melange.emit`` Stanzas
+---------------------------------------------------------------------
+
+It is recommended to define a single ``melange.emit`` stanza for each project,
+or if that is not possible, at least minimize the number of ``melange.emit``
+stanzas defined.
+
+The reason is that using multiple ``melange.emit`` stanzas will lead to multiple
+copies of the JavaScript files generated for library code. For example, if
+there are two ``melange.emit`` stanzas defined as follows:
+
+.. code:: dune
+
+  (melange.emit
+   (target app1)
+   (libraries foo))
+
+  (melange.emit
+   (target app2)
+   (libraries foo))
+   
+The JavaScript artifacts for library ``foo`` will end up duplicated in the
+``_build`` folder, both in ``_build/default/app1`` and ``_build/default/app2``.
+
+This can have unexpected impact on bundle size when using tools like Webpack or
+Esbuild, as these tools will not be able to see shared library code as such,
+as it would be replicated across the paths of the different stanzas
+``target`` folders.
+
+
+Faster Builds with ``subdir`` and ``dirs`` Stanzas
+--------------------------------------------------
+
+Melange libraries might be installed from the ``npm`` package repository,
+together with other JavaScript packages. To avoid having Dune inspect
+unnecessary folders in ``node_modules``, it is recommended to explicitly
+include those folders that are relevant for Melange builds.
+
+This can be accomplished by combining :ref:`subdir` and :ref:`dune-subdirs`
+stanzas in a ``dune`` file and by co-locating this file together with the
+``node_modules`` folder. The :ref:`dune-vendored_dirs` stanza can be used as
+well to avoid warnings in Melange libraries during the application build.
+
+.. code:: dune
+
+  (subdir
+   node_modules
+   (vendored_dirs reason-react)
+   (dirs reason-react))
+
+Simple Project
+==============
+
+Let's look at building a simple project with Melange and Dune.
+
+First, make sure that the :ref:`dune-project` file contains a
+``(using melange 0.1)`` stanza:
+
+.. code:: dune
+
+  (lang dune 3.8)
+  (using melange 0.1)
+
+Next, write a :ref:`dune<dune-files>` file with a :ref:`melange-emit` stanza:
+
+.. code:: dune
+
+  (melange.emit
+   (target output))
+
+Finally, add a ``.ml`` source file to build:
+
+.. code:: bash
+
+  echo 'Js.log "hello from melange"' > hello.ml
+
+Now, run ``dune build output/hello.js``. Dune produces the following
+structure once it has finished the build:
+
+.. code::
+
+  .
+  ├── _build
+  │   └── default
```

Comment:
this is wrong. should be `_build/default/output/hello.js`. The output dir is missing

Reply:
added in https://github.com/ocaml/dune/pull/7477/commits/b43dfb8f9502945f6de2e6dce551353a575388c7